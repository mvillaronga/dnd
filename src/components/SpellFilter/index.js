import React, { useRef } from 'react'

const SpellFilter = () => {
    const entry = useRef()

    const props = {

    }

    return (
        <div>
            Filter&nbsp;
            <input type="text" ref={entry} {...props} />
        </div>
    )
}

export default SpellFilter