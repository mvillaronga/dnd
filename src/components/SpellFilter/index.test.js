import React from 'react';
import SpellFilter from '../SpellFilter';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('Spell Filter tests', () => {
  let filter = ''
  const onFilterChange = jest.fn(val => filter = val)
  const onFilterClear = jest.fn( () => filter = '' )

  const props = {
    filter: filter,
    onFilterChange: onFilterChange,
    onFilterClear: onFilterClear
  }
  const wrapper = () => mount(
    <SpellFilter {...props} />
  )

  it('should render properly', () => {
    const component = renderer.create(
      <SpellFilter {...props} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render properly with a filter', () => {
    const component = renderer.create(
      <SpellFilter {...props} filter='set' />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should handle changes to the filter', () => {
      const comp = wrapper();
      const ip = comp.find('input').first()
      ip.getDOMNode().value = 'test'
      ip.simulate('change')
      expect(onFilterChange).toHaveBeenCalled()
      expect(filter).toEqual('test')
  })

  it('should handle the clear button being clicked', () => {
    props.filter = 'filt'
    const comp = wrapper();
    comp.find('button').first().simulate('click')
    expect(onFilterClear).toHaveBeenCalled()
    expect('').toEqual(filter)
  })
})