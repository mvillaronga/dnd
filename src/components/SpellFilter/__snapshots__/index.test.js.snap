// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Spell Card Rendering 1`] = `
<div>
  FilterÂ 
  <input
    type="text"
  />
</div>
`;
